v3: Remove optional tab autocompletion feature

medikoo opened the pull request:
Reasoning:
- Performance is very poor due to slow command startup time
- `tabtab` package is no longer maintained and shares security issues
- Feature usage is relatively low (telemetry data shows 7% of users has the autocompletion on)

Closes: #8473

Addresses #10182

pgrzesik reviewed (approved):
Looks great, good call 👍

mnapoli commented:
Should we also delete this documentation page: https://www.serverless.com/framework/docs/providers/aws/cli-reference/config-tabcompletion ?

medikoo commented:
> Should we also delete this documentation page: https://www.serverless.com/framework/docs/providers/aws/cli-reference/config-tabcompletion ?

Ouch, indeed I missed that. I've fixed it and updated merged commit
